Libraries/Components/WebView/WebView.ios.js

```diff
@@ -462,6 +462,38 @@ var WebView = React.createClass({
   },
 
   /**
+   * Evaluate JavaScript on the current page.
+   */
+  evaluateJavaScript: function(
+    script: string,
+    callback?: ?(error: ?Error, result: ?string) => void,
+  ): Promise {
+    return new Promise((resolve, reject) => {
+      var escaped = JSON.stringify(script).replace(/\u2028/g, '\\u2028').replace(/\u2029/g, '\\u2029');
+      var wrapped = 'JSON.stringify(eval(' + escaped + '))';
+
+      RCTWebViewManager.evaluateJavaScript(this.getWebViewHandle(), wrapped, function(error, result) {
+        if (!error) {
+          if (result === '') {
+            error = new Error("Error evaluating script.")
```

Comment:
Oh -- I thought you were suggesting we try to get the actual JS error from the page. Yeah, we could create this general error in Objective-C instead, though I don't think it matters much either way.

If the result is an empty string, we'll return the JSON stringification of that empty string (i.e., `"\"\""`). Since results are always stringified, `JSON.parse()` will always succeed for non-error cases, and `""` isn't valid JSON.